### PR TITLE
storage: fix flaky test TestHeartbeatHealth

### LINKED ISSUE #7410: circleci: failed tests: TestHeartbeatHealth
The following test appears to have failed:

[#19491](https://circleci.com/gh/cockroachdb/cockroach/19491):

```
--- SKIP: TestClockOffsetMetrics (0.01s)
	clock_offset_test.go:163: 
=== RUN   TestHeartbeatCB
--- PASS: TestHeartbeatCB (0.15s)
=== RUN   TestHeartbeatHealth
--- FAIL: TestHeartbeatHealth (15.27s)
	<autogenerated>:32: rpc/context_test.go:125: condition failed to evaluate within 15s: rpc/context_test.go:122: expected 127.0.0.1:58535 to be healthy
=== RUN   TestOffsetMeasurement
--- PASS: TestOffsetMeasurement (0.31s)
=== RUN   TestFailedOffsetMeasurement
--- PASS: TestFailedOffsetMeasurement (0.12s)
=== RUN   TestRemoteOffsetUnhealthy
I160622 21:44:14.246232 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.246521 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.246705 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.252815 http2_client.go:919  transport: http2Client.notifyError got notified that the client transport was broken EOF.
--
--- PASS: TestRemoteOffsetString (0.01s)
=== RUN   TestHeartbeatReply
--- PASS: TestHeartbeatReply (0.02s)
=== RUN   TestManualHeartbeat
--- PASS: TestManualHeartbeat (0.01s)
FAIL
FAIL	github.com/cockroachdb/cockroach/rpc	17.252s
=== RUN   TestGetCertificateUser
--- PASS: TestGetCertificateUser (0.02s)
=== RUN   TestAuthenticationHook
--- PASS: TestAuthenticationHook (0.01s)
=== RUN   TestGenerateCerts
--- PASS: TestGenerateCerts (0.11s)
=== RUN   TestUseCerts
I160622 21:44:28.268027 storage/engine/rocksdb.go:143  opening in memory rocksdb instance
W160622 21:44:28.274284 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
W160622 21:44:28.283173 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by d4l3k**
Fixes #7410.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7415)

### CODE REVIEW

File: rpc/context_test.go [hunk 1]
```diff
@@ -109,20 +109,41 @@ func TestHeartbeatHealth(t *testing.T) {
...
+	stopHeartbeats := sendHeartbeats()
 	util.SucceedsSoon(t, func() error {
 		if !clientCtx.IsConnHealthy(remoteAddr) {
 			return util.Errorf("expected %s to be healthy", remoteAddr)
 		}
 		return nil
 	})
+	stopHeartbeats()
 
 	// Should no longer be healthy after later heartbeats fail.
```
Comment (tschottdorf) [inline rpc/context_test.go]: s/later heartbeats fail/heartbeating stops/

**Comment by tschottdorf** [comment]
LGTM

**Comment by d4l3k** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[rpc/context_test.go, line 148 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-KKuK1VGSl53OYVd06iN-r1-148:-KKuMuUqacqe9e8RvCqN:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/cab09087fe2ec0bbc9d13b4d6651b9b31490ab48/rpc/context_test.go#L148)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> s/later heartbeats fail/heartbeating stops/
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[rpc/context_test.go, line 128 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-KKuepzcK0Y3iJ1OK-uX:-KKuepzdnsvMWBHDwNiL:1605423663) ([raw file](https://github.com/cockroachdb/cockroach/blob/28c689f5aea9d84830cddbb31b72443e77674592/rpc/context_test.go#L128)):*
> ```Go
> 					return
> 				case heartbeat.ready <- struct{}{}:
> 				default:
> ```

what's the point of this default clause and the sleep?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-:-KKuf54cQU05JFUZWsE9:1417088721)*

**Comment by d4l3k** [comment]
*[rpc/context_test.go, line 128 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-KKuepzcK0Y3iJ1OK-uX:-KKxqUzxYy1az3N0w8pq:-514497717) ([raw file](https://github.com/cockroachdb/cockroach/blob/28c689f5aea9d84830cddbb31b72443e77674592/rpc/context_test.go#L128)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> what's the point of this default clause and the sleep?
</details>
Just to keep it from constantly spinning since we really should only need to send a heartbeat every 1ms.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[rpc/context_test.go, line 128 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-KKuepzcK0Y3iJ1OK-uX:-KKxr4loIFNLtuMKdtx7:-1790059002) ([raw file](https://github.com/cockroachdb/cockroach/blob/28c689f5aea9d84830cddbb31b72443e77674592/rpc/context_test.go#L128)):*
<details><summary><i>Previously, d4l3k (Tristan Rice) wrote…</i></summary>
> Just to keep it from constantly spinning since we really should only need to send a heartbeat every 1ms.
</details>
Doesn't the client side take care of that? Seems to me that this sleep doesn't do anything.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415)*

**Comment by d4l3k** [comment]
*[rpc/context_test.go, line 128 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415#-KKuepzcK0Y3iJ1OK-uX:-KKxrRiDPlsdCWLTPrCD:1491125807) ([raw file](https://github.com/cockroachdb/cockroach/blob/28c689f5aea9d84830cddbb31b72443e77674592/rpc/context_test.go#L128)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Doesn't the client side take care of that? Seems to me that this sleep doesn't do anything.
</details>
Ohhhhh... I see what you mean. Yeah, we can get rid of that. I'll send another PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7415)*

**Pull request merged**